src/mem/chunkmap.h

```diff
@@ -163,6 +180,24 @@ namespace snmalloc
       return get(address_cast(p));
     }
 
+    /**
+     * Some architectures (notably, CHERI) need the ability to rederive an
+     * internal, privileged pointer from a pointer given out as a result of
+     * allocation.  This provides such a hook.  On architectures where
+     * pointers are just integers, this is a no-op that should always be
+     * inlined into nothing.
+     *
+     * The template paramter controls whether the returned pointer points
+     * to the same place as the argument or to the beginning of the chunkmap
+     * region containing the argument.
+     */
+    template<bool offset = true>
+    SNMALLOC_FAST_PATH static void* getp(void* p)
```

Comment:
Please give this a more informative name.  I came across the uses of this before the definition in the review and had no idea what `p` was or why I was getting it from the chunk map.  I'd suggest `rederrive_pointer` or similar.

Reply:
Sure.